src/app/page/Recipient/Details.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { ReactNode, useState, useContext } from "react";
+import { Typography, Button, Box, Divider, TextField } from "@material-ui/core";
+import styled from "styled-components";
+
+import Snackbar from "../../component/Snackbars";
+import { MissionInterface, MissionStatus as Status } from "../../model/schema";
+import { Mission, useOrganization } from "../../model";
+
+export default function Details({ mission }: { mission: MissionInterface }) {
+  const org = useOrganization();
+  return (
+    <>
+      <Divider />
+      <Box margin="1rem">
+        {mission.deliveryType === "delivery" ? (
+          <>
+            <DetailSection
+              header="Address"
+              content={
+                <Typography variant="body1" gutterBottom>
+                  {mission.deliveryLocation.address}
+                </Typography>
+              }
+            />
+            <DropOffInstructions mission={mission} />
+          </>
+        ) : (
+          <>
+            <DetailSection
+              header="Pick Up Location"
+              content={
+                <>
+                  <Typography variant="body1">{org?.location?.label}</Typography>
+                  <Typography variant="body1">{org?.location?.address}</Typography>
+                </>
+              }
+            />
+            <DetailSection
+              header="Pick Up Time"
+              content={
+                <Typography variant="body1">Sunday morning between 8:00am–11:00am</Typography>
+              }
+            />
+          </>
+        )}
+        {org?.contactEmail && (
+          <DetailSection
+            header="Need Help?"
+            content={
+              <Typography variant="body1" gutterBottom>
+                Contact your organizer at{" "}
+                <a href={`mailto:${org?.contactEmail}`}>{org?.contactEmail}</a>
+              </Typography>
+            }
+          />
+        )}
+      </Box>
+    </>
+  );
+}
+
+function DetailSection({ content, header }: { content: ReactNode; header: ReactNode }) {
+  return (
+    <Box marginBottom="1rem">
+      {React.isValidElement(header) ? header : <Typography variant="h5">{header}</Typography>}
+      <Box>{content}</Box>
+    </Box>
+  );
+}
+
+function DropOffInstructions({ mission }: { mission: MissionInterface }) {
+  const [isEditing, setIsEditing] = useState(false);
+  const [deliveryNotes, setDeliveryNotes] = useState(mission.deliveryNotes);
+
+  const snackbar: any = useContext(Snackbar.Context.SnackbarContext);
+
+  function handleEdit() {
+    if (!isEditing) {
+      setIsEditing(true);
+      return;
+    }
+
+    Mission.update(mission.uid, { deliveryNotes })
+      .then(() => {
+        setIsEditing(false);
+      })
+      .catch((e) => {
```

Comment:
Do we want to provide `e.message` for any reason to the user or organizer? Is it possible Mission.update would have any valuable info that they could resolve?

Reply:
I wish we had a logging service... I can `console.error` it but I don't think customers need to see the raw message